From 935691d0f018609203d40e6e207b56cbe8e8e9c0 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 08 Dec 2021 09:15:41 +0200
Subject: [PATCH] [SW-63186] gaudi2: security: set axprot for hmmu

Set axprot for hmmu rtr ifs as secure access level.

Change-Id: I3759b55f6d7d11dbee139ac815d57ec20dcc90b6
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 674f6c6..8640316 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -35,6 +35,7 @@
 #include <dcore0_vdec0_mstr_if_core_lbw_regs.h>
 #include <rot0_mstr_if_core_hbw_regs.h>
 #include <rot0_mstr_if_core_lbw_regs.h>
+#include <dcore0_hmmu0_mstr_if_core_hbw_regs.h>
 
 /*
  * instances_num : Number of instances of RR to program.
@@ -197,6 +198,7 @@
 enum preboot_sec_axprot_regs {
 	AXPROT_VDEC,
 	AXPROT_ROT,
+	AXPROT_HMMU,
 };
 
 #define DCORE_DIST	0x200000
@@ -921,6 +923,10 @@
 #define ROT_RTR_INST_NUM	1
 #define ROT_INST_NUM		2
 #define ROT_INST_DIST		0x10000
+#define HMMU_RTR_INST_NUM	NUM_OF_DCORES
+#define HMMU_RTR_INST_DIST	0x10000
+#define HMMU_INST_NUM		NUM_OF_DCORES
+#define HMMU_INST_DIST		DCORE_DIST
 
 static struct axprot_regs axprot_reg[] = {
 	[AXPROT_VDEC] = {
@@ -938,6 +944,14 @@
 		.dcore_num = ROT_INST_NUM,
 		.inst_dist = 0,
 		.inst_num = ROT_RTR_INST_NUM,
+	},
+	[AXPROT_HMMU] = {
+		.base_reg = mmDCORE0_HMMU0_MSTR_IF_CORE_HBW_AXPROT_OVRD,
+		.value = AXPROT_SEC_VAL,
+		.dcore_dist = HMMU_INST_DIST,
+		.dcore_num = HMMU_INST_NUM,
+		.inst_dist = HMMU_RTR_INST_DIST,
+		.inst_num = HMMU_RTR_INST_NUM,
 	},
 };
 
@@ -978,6 +992,13 @@
 	preboot_axprot_regs_init(AXPROT_ROT);
 }
 
+static void preboot_secure_hmmu(void)
+{
+	/* Configure hmmu to secure level */
+	/* HBW AXPROT */
+	preboot_axprot_regs_init(AXPROT_HMMU);
+}
+
 void preboot_security_init(void)
 {
 	preboot_secure_pcie();
@@ -988,6 +1009,7 @@
 		hl_dev_prot_bits_init();
 		preboot_secure_vdec();
 		preboot_secure_rot();
+		preboot_secure_hmmu();
 		preboot_secure_lbw();
 		preboot_secure_hbw();
 		preboot_puts("Security enabled.\n");
